repos/fdbt-site/src/pages/viewPassengerTypes.tsx

```diff
@@ -1,74 +1,125 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useState } from 'react';
 import { BaseLayout } from '../layout/Layout';
-import { PassengerType, NextPageContextWithSession, GroupPassengerType } from '../interfaces';
-import { getAndValidateNoc, sentenceCaseString } from '../utils';
+import { SinglePassengerType, NextPageContextWithSession, GroupPassengerType } from '../interfaces';
+import { getCsrfToken, getAndValidateNoc, sentenceCaseString } from '../utils';
 import { getPassengerTypesByNocCode } from '../data/auroradb';
 import SubNavigation from '../layout/SubNavigation';
+import DeleteConfirmationPopup from '../components/DeleteConfirmationPopup';
 
 const title = 'Passenger types';
 const description = 'View and edit your passenger types.';
 
 interface PassengerTypeProps {
-    passengerTypes: PassengerType[];
-    passengerTypeGroups: GroupPassengerType[];
+    csrfToken: string;
+    singlePassengerTypes: SinglePassengerType[];
+    groupPrssengerTypes: GroupPassengerType[];
 }
 
-const ViewPassengerTypes = ({ passengerTypes, passengerTypeGroups }: PassengerTypeProps): ReactElement => (
-    <BaseLayout title={title} description={description} showNavigation={true}>
-        <div className="govuk-width-container">
-            <main className="govuk-main-wrapper">
-                <div className="govuk-grid-row">
-                    <div className="govuk-grid-column-one-third">
-                        <SubNavigation />
-                    </div>
+const ViewPassengerTypes = ({
+    singlePassengerTypes,
+    groupPrssengerTypes,
+    csrfToken,
+}: PassengerTypeProps): ReactElement => {
+    const [popUpState, setPopUpState] = useState({
+        isVisible: false,
+        passengerTypeName: '',
+        isGroup: false,
+    });
+
+    const deleteActionHandler = (name: string, isGroup: boolean): void => {
+        const previousState = { ...popUpState };
+
+        previousState.isVisible = true;
+        previousState.passengerTypeName = name;
+        previousState.isGroup = isGroup;
+
+        setPopUpState({ ...previousState });
```

Comment:
Minor: I think our convention is to do this kind of thing with full object spreading: 
```
setPopUpState(
{
 ...previousState,
isVisible: true,
passengerTypeName: name,
...
 }
);
```